proof-producer/bin/proof-producer/CMakeLists.txt

```diff
@@ -47,6 +47,7 @@ function(setup_proof_generator_target)
     if(ARG_ADDITIONAL_DEPENDENCIES)
         foreach(lib IN LISTS ARG_ADDITIONAL_DEPENDENCIES)
             target_link_libraries(${ARG_TARGET_NAME}-lib INTERFACE ${lib})
+            target_compile_definitions(${ARG_TARGET_NAME}-lib INTERFACE MULTITHREADED)
```

Comment:
check if `lib` is `parallel-crypto3`.
Generally single-thread version also can has some additional dependency